src/server/game/Entities/GameObject/GameObject.cpp

```diff
@@ -2227,6 +2235,27 @@ bool GameObject::IsLootAllowedFor(Player const* player) const
     return true;
 }
 
+GameObject* GameObject::GetLinkedTrap()
+{
+    uint32 linkedEntry = GetGOInfo()->GetLinkedGameObjectEntry();
+
+    if (!linkedEntry)
+        return nullptr;
+
+    GameObject* linkedTrap = nullptr;
+
+    if (m_linkedTrap.Empty())
```

Comment:
The issue with that approach is that we have no control over which gameobject is spawned first, if the trap or the "master".

Am I missing something?